### ISSUE TITLE
How to test onRowClicked in React Testing Library?

### ISSUE THREAD

**Issue by simkessy** [ag-grid/ag-grid#4621]
I've been able to get the grid to render in my test by waiting for the onready:

```
const ensureGridApiHasBeenSet = () => {
  // eslint-disable-next-line func-names
  return new Promise(function(resolve) {
    (function waitForGridReady() {
      // screen.debug();
      const gridReadyElement = screen.getByTestId('groups');
      if (gridReadyElement.dataset.gridReady === 'true') {
        return resolve();
      }
      setTimeout(waitForGridReady, 100);
    })();
  });
};
```

Then in my tests I do: 

```
  beforeEach(async () => {
    component = renderComponent(defaultProps);
    await waitFor(() => ensureGridApiHasBeenSet());
  });

  afterEach(() => {
    component.unmount();
  });
```

I pass a mock to the `onRowClicked` property in my test and write this test: 

```
  it('trigger click event', () => {
    const rows = screen.getAllByRole('row');
    rows.forEach(row => {
      fireEvent.click(row);
    });

    expect(rowClickMock).toHaveBeenCalled();
  });
```

None of the rows getting clicked are triggering the mock

**Comment by makinggoodsoftware** [comment]
Hi,

 

This ticket has been flagged as managed-by-community/waiting-for-repro for a while now and there has not been any activity, to help us tidy up we are going to close it.

 

Thanks

**Comment by elliot-at-looker** [comment]
+1 I am getting this too! Was there ever a solution or work around identified?

**Comment by apoorva-srivastav** [comment]
I am also getting the same issue. Can this be re-opened @makinggoodsoftware ?

**Comment by devprachi** [comment]
I am also getting the same issue. Couldn't find any solution.

**Comment by zoheil-ag** [comment]
Hi all,

As this issue is quite old and there had never been a reproducible scenario provided alongside it I would suggest to open a new issue with a plunker to replicate the problem you are experiencing.

Depending on the actuality of the issue, we can then either look into it here at AG-Grid or our lovely community can provide support. This will also be the best way to get more eyes on it from the community.

Kind regards,
Zoheil

**Comment by simkessy** [comment]
There is no way to test AG Grid with RTL. Im not sure why they won't just come out and say this. Support for this tool is garbage.

**Comment by kiril-matev** [comment]
Hello @simkessy,

Thanks for reporting this.

AG Grid supports testing libraries and we're keen to investigate any issues reported against this. 

In this thread, you're reporting potential issues but without the necessary context for us to investigate them. This is making it difficult for us to assist as we need the complete context of use in a live sample that we can run locally to reproduce the issue. 

Please provide github repos which reproduces the issues as per the bug submission template. We'd be keen to look into it.

We're looking forward to your response.

**Comment by simkessy** [comment]
This is 2 years old, I don't care about this issue anymore. Close it if you want. Im just pointing out that React Testing Library is the biggest testing library in JavaScript and yet AgGrid has provided absolutely zero support and documentation on how to effectively test their components using it.

**Comment by kiril-matev** [comment]
I see your point. I just want to emphasise that we're keen to ensure that AG Grid supports widely used testing libraries and committed to improving the quality of AG Grid.

When there are issues to be reported, we rely on the AG Grid community to report them using code samples as I mentioned. Only once an issue is properly reported, we will investigate it and make product improvements if needed so AG Grid can be used effectively with the React Testing Library.

So whenever you have issues to report, please follow the bug submission template we provide and we'd be happy to look into those.

**Comment by simkessy** [comment]
>  I just want to emphasise that we're keen to ensure that AG Grid supports widely used testing libraries 

I don't believe you. 

https://github.com/ag-grid/ag-grid/issues/4111

**Comment by kiril-matev** [comment]
@simkessy Thank you for this response.

I understand you don't have to believe me and I'm not expecting that. 

Simply send us sample projects to help us reproduce any issues - we will investigate them and provide our guidance. This goes for any issues, not just related to support for testing libraries.

There may be issues we can fix or workaround. We may not be able to resolve every issue but you'll have our assessment and a clear indication of the scenarios where you can use AG Grid with testing libraries, so you can plan your testing better.

**Comment by StephenCooper** [comment]
I have followed up on this and updated the following example repo with a test that shows examples of clicking on cells and rows using RTL. 

I have not found any specific issues while using RTL so maybe I am missing something?

https://github.com/seanlandsman/ag-grid-react-hook-testing

I will be updating our documentation page on React Testing to include a section on RTL for the next release.

**Comment by AG-Zoheil** [comment]
As this has been open without any further interaction for a while, I will be closing this issue.
Please note in line with **AG-9210** we will be updating our testing documentation across several pages/examples.

If the issue has already been resolved by other means, you can ignore this. If however, you are still in need of support you can choose to raise the issue again for the AG Grid community to support.

Kind regards,
Zoheil
